tools/rosbag/src/record.cpp

```diff
@@ -279,8 +280,22 @@ rosbag::RecorderOptions parseOptions(int argc, char** argv) {
     return opts;
 }
 
+/**
+ * Handle SIGINT and SIGTERM to allow the recorder to cleanup by requesting a shutdown.
+ * \param signal
+ */
+void signal_handler(int signal)
+{
+  (void) signal;
+  ros::requestShutdown();
+}
+
 int main(int argc, char** argv) {
-    ros::init(argc, argv, "record", ros::init_options::AnonymousName);
+    ros::init(argc, argv, "record", ros::init_options::NoSigintHandler | ros::init_options::AnonymousName);
+
+    // handle SIGINT and SIGTERM signals
+    signal(SIGINT, signal_handler);
```

Comment:
The registered custom signal handler is the same as the default for `SIGINT`. Therefore I think it should be removed from this patch together with `ros::init_options::NoSigintHandler`.

Reply:
Agree, done.